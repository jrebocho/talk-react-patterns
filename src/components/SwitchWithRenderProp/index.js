import React from 'react'
import StatelessSwitch from '../StatelessSwitch'
import BlueClickRenderProp from '../BlueClickRenderProp'

const SwitchWithRenderProp = () => (
  <BlueClickRenderProp render={on => <StatelessSwitch on={on} />} />
)

export default SwitchWithRenderProp
