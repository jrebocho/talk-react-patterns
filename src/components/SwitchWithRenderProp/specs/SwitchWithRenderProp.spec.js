import React from 'react'
import { shallow } from 'enzyme'
import SwitchWithRenderProp from '../../SwitchWithRenderProp'

describe('<SwitchWithRenderProp />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SwitchWithRenderProp />)
    expect(wrapper).toMatchSnapshot()
  })
})
