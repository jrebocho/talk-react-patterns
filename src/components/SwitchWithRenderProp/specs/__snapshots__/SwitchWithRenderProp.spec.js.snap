// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchWithRenderProp /> renders correctly 1`] = `
<BlueClickRenderProp
  render={[Function]}
/>
`;
